### DISCUSSION TITLE
Latest iOS bitwarden app problems

### DISCUSSION THREAD

**Discussion by yry-dev** [dani-garcia/vaultwarden#5012] [General]
I am self hosting with vaultwarden and I can still access my instance from my browser plugins no problem but the iOS app throws "An Error Has Occured" immediately before I've even entered a password.  Has anyone else seen this behavior?  I confirmed in the nginx proxy logs and the vault warden logs that the iOS device never even hits the server.

**Comment by bbrendon** [comment]
yes. upgrade your server.

**Reply by LeehamElectronics** [reply]
I'm having the same problem, I am using latest vaultwarden/server:latest

**Reply by LeehamElectronics** [reply]
Apologies, I updated again, rm container and tried again and it now works fine. Thanks

**Comment by yry-dev** [comment]
I literally changed nothing and it started working for me again this morning?  I am going to assume it was an update somewhere that fixed it

**Comment by dr400** [comment]
Hi running latest 1.32.2 with IOS 17.7.
IOS application has been removed (completely ?), phone switched offand back on. IOS application re-installed. But No joy ! Still the "ann error has occured".

Any suggestion ?

**Reply by BlackDex** [reply]
Yes, use the `testing` tagged version.

**Comment by ShianHui** [comment]
Hi all,

A noob here.  I am experiencing similar symptom. 

I run a self-hosted Vaultwarden docker on an Unraid server with Cloudflared Tunnel. I tunnel all my traffic for Vaultwarden through the Cloudflared Tunnel. The domain name is hosted on Cloudflared pointing to the tunnel, which points to my docker container on my Unraid.

Everything works like a charm using web-browser and browser plugins! 

However, when I install Bitwarden app on my iPhone, the app kept complaining "An error has occured." without much details. 

I have tried the testing version 1.32.5-da3701c0 as well as the latest version 1.32.5. Both had the same error. Reinstalled the Bitwarden app on my iPhone multiple times. 

Anyone can give me some clue where to find information to troubleshoot? Much appreciated.

**Reply by ShianHui** [reply]
Folks, I just got the iOS app to work with my above setup.

Steps I took to resolve:
1. Keep Vaultwarden version to 1.32.5
2. Delete and reinstall the Bitwarden app on the Apple Mobile Phone.
3. Attempt to login to "bitwarden.com". It will fail but this action seems to trigger something on the client. NO CLUE WHAT VOODOO that does.
4. Set the "Accessing" to "self-hosted" and set my vaultwarden hostname.
5. Magically works without encountering the error anymore.

Like I say, it's like voodoo or magic. I have no clue what the client did but the above steps have been tested twice to work. If I reinstalled the app, immediately set to "Self-hosted" mode, I will get the error. Anyway, I am over the moon that it works. Not bothered by a little workaround! Cheers!

**Reply by burgich** [reply]
Thanks alot! After a day of searching, finally the solution...
I had exactly the same problem with the new Bitwarden Android app that has just been released. 
I then followed your instructions here and was able to log in!
As you say, voodoo magic!

**Reply by felix2911-git** [reply]
This workaround worked perfectly on my old iPhone, on my new iPhone I keep getting this error. Does anybody has a solution for this?

**Reply by ShianHui** [reply]
I happen to have a spare iPhone. Reset it to factor default to try it. Steps I took:

1. Check that existing iOS Bitwarden (v2025.12.0 (2787)) and Docker Vaultwarden (v1.35.2) version work with my current iPhone iOS (v26.2).
2. Install Bitwarden from Apple store. 
3. Select Login using fictious account name to "bitwarden.com". Wait for logon to fail and report error.
4. Select login again this time with correct account name to my own hosted Vaultwarden server  with a Public IP and Domain name. 
5. Under Server url, enter: https://<your public hostname>. Keep custom config empty. Set "remember me"
6. Use my username and password. Login.
Successfully logged in. 

Past experience (though did not test this round):
1. Should be https, else doesn't seem to work.
2. I believe I tried IP before, couldn't recall accurately it worked. I recalled I had some problems with IP.

Try my steps. They are still working for me even with setting up a new phone. Good luck! Wishing you success!

**Comment by Manah7** [comment]
Same problem here, but the ShianHui's workaround do not work for me. I am using the latest vaultwarden image from docker repositories.

**Reply by eboman** [reply]
Same for me after logging in with my mfa code I even get an confirmation email and user logged in , but keeps throwing the error.

**Reply by BlackDex** [reply]
Use the flight recorder feature to get more details 
https://bitwarden.com/help/flight-recorder/

**Reply by eboman** [reply]
After an successful /api/config I see this error

```
2026-01-17T10:45:40+01:00: Error: missingMasterPasswordUnlockData
0   Bitwarden                           0x0000000104ea6b0c Bitwarden + 27404
1   BitwardenShared                     0x000000010558ec48 block_destroy_helper + 44156
2   BitwardenShared                     0x000000010558e285 block_destroy_helper + 41657
3   BitwardenShared                     0x000000010532f329 objectdestroy.5Tm + 11633
4   BitwardenKit                        0x0000000105d405a5 __swift_instantiateGenericMetadata + 1841
5   BitwardenKit                        0x0000000105d47e75 __swift_mutable_project_boxed_opaque_existential_1 + 1421
6   BitwardenKit                        0x0000000105d408dd __swift_instantiateGenericMetadata + 2665
7   BitwardenKit                        0x0000000105dfaa81 objectdestroy.13Tm + 905
8   BitwardenShared                     0x00000001053328b1 objectdestroy.4Tm + 12785
9   BitwardenShared                     0x00000001052ed509 __swift_memcpy1_1 + 5525
10  BitwardenKit                        0x0000000105d4aded __swift_memcpy1_1 + 2033
11  BitwardenKit                        0x0000000105d405a5 __swift_instantiateGenericMetadata + 1841
12  BitwardenKit                        0x0000000105daa901 objectdestroy.4Tm + 28533
13  BitwardenKit                        0x0000000105d408dd __swift_instantiateGenericMetadata + 2665
14  libswift_Concurrency.dylib          0x000000019ffdd7b1 5B1111B6-2106-3AE3-BDEB-34A8B033CD74 + 444337
2026-01-17T10:45:45+01:00
```

**Reply by BlackDex** [reply]
Are you running the latest Vaultwarden??

**Reply by eboman** [reply]
Yes just did an forced pull using the latest tag. Is there an version number in the gui to confirm ?

**Reply by BlackDex** [reply]
Be sure to check the `/admin/diagnostics` page to verify.

**Reply by eboman** [reply]
Ok will need to check the page still gives version 1.34.3. Something seems off for my image updates.

**Reply by BlackDex** [reply]
Pulling an image doesn't update it automatically. You need to either re-create the container or run `docker compose up -d`

**Reply by eboman** [reply]
Ok seems portainer did not pull new images when updating. Got an new image now and all good. Thanks for the support
